sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -92,7 +92,7 @@ statement
     | CREATE TEMPORARY? FUNCTION qualifiedName AS className=STRING
         (USING resource (',' resource)*)?                              #createFunction
     | DROP TEMPORARY? FUNCTION (IF EXISTS)? qualifiedName              #dropFunction
-    | EXPLAIN explainOption* statement                                 #explain
+    | EXPLAIN (LOGICAL | FORMATTED | EXTENDED | CODEGEN)? statement    #explain
```

Comment:
: ) Based on my understanding, it outputs optimized logical plan. https://issues.apache.org/jira/browse/HIVE-4812 Thus, I assume we do not need to support `Explain Logical`, which is already included in our current `Explain Extended`. Let me know if we want to support it. This is an undocumented Hive feature. 

Compared with Hive, we are not following what Hive does. For example, `Explain Extended` only outputs the AST and Physical plan.